### PR TITLE
Update to py311.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
This is expected to die on rubin-sim.

**Comment by erykoff** [comment]
This would be a good time to bump our compiler versions as well on dev.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -142,6 +144,7 @@ outputs:
         - idds-client >=0.11.0
         - idds-doma >=0.11.0
         - iminuit >=2
+        - importlib_resources
```
Comment (timj) [inline recipe/meta.yaml:147]: We don't need this for python 3.11 because `importlib.resources` has the good API.

**Review by timj — APPROVED** [review]

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by erykoff** [comment]
rubin-sim is now available for py311.

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by erykoff** [comment]
We don’t want to leave libcxx completely unpinned though. If we want to maintain binary installation compatibility of our eups packages we need to pick one (most recent now) and stick with it.

**Comment by ktlim** [comment]
Interestingly, it was only pinned for `[osx]` before.  Does the `sysroot_linux-64` pin have the same effect for that platform?

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Comment by erykoff** [comment]
Re: rubin-sim for aarch64, that at least requires openorb for aarch64.  Considering how difficult the osx-arm64 build was, I don't know what that entails... but either the hard work has been done (easy) or another heavy lift.  I guess I can give it a try, and if it works out of the box, great!  But if it doesn't I'm not going to spend any more time on it.

**Comment by erykoff** [comment]
Meanwhile, are we planning to update our numpy pin?

**Comment by erykoff** [comment]
rubin-sim is now available for aarch64.

**Comment by ktlim** [comment]
Superseded by #163.

**Pull request closed without merging**